miden-lib/asm/kernels/transaction/lib/memory.masm

```diff
@@ -622,37 +606,73 @@ end
 # ACCOUNT DATA
 # -------------------------------------------------------------------------------------------------
 
-#! Returns a pointer to the account data.
+#! Sets the offset of the current account data to the native account (2048).
+#!
+#! Stack: []
+#! Output: [native_account_data_ptr]
+#!
+#! Where:
+#! - native_account_data_ptr is the memory address at which the native account data is stored.
+export.set_current_account_data_offset_to_native_account
+    push.NATIVE_ACCOUNT_DATA_PTR
+    push.CURRENT_ACCOUNT_DATA_OFFSET mem_store
+end
+
+#! Returns the offset of the current account data.
 #!
 #! Stack: []
 #! Output: [ptr]
 #!
 #! Where:
 #! - ptr is the memory address at which the account data begins.
-export.get_acct_data_ptr
-    push.ACCT_DATA_SECTION_OFFSET
+export.get_current_account_data_offset
+    push.CURRENT_ACCOUNT_DATA_OFFSET mem_load
+end
```

Comment:
Similar to the previous comment: this should probably be `get_current_account_data_ptr`. Also relevant in a few other places here.